### PR TITLE
Colin/better build order

### PR DESCRIPTION
**Pull request by colin-higgins**
Move c++ projects to top of solution to have them build before samples.
Replace inconsistent OutputRoot variable.
Only copy relevant profiler subdirectory.

### CODE REVIEW

File: samples/Directory.Build.props [hunk 1]
```diff
@@ -17,11 +17,13 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Content Include="..\..\src\Datadog.Trace.ClrProfiler.Native\bin\**"
+    <None Include="..\..\src\Datadog.Trace.ClrProfiler.Native\bin\$(Configuration)\\$(Platform)\\**"
```
Comment (lucaspimentel) [inline samples/Directory.Build.props]: You can probably change the end here to `**\*.dll` so it only copies the dll files.

Comment (colin-higgins) [reply samples/Directory.Build.props]: what about `*.so`?

Comment (lucaspimentel) [reply samples/Directory.Build.props]: The samples (and any profiled app) only need `Datadog.Trace.ClrProfiler.Native.dll`.

Comment (lucaspimentel) [reply samples/Directory.Build.props]: Never mind me. I forgot about Linux.

**Pull request merged**